client/src/components/DataFiles/DataFilesProjectMembers/DataFilesProjectMembers.js

```diff
@@ -0,0 +1,238 @@
+      className: 'project-members__cell',
+      Cell: el => (
+        <>
+          {mode === 'addremove' && el.row.original.access !== 'owner' ? (
+            <Button
+              onClick={e => onRemove(el.row.original)}
+              color="link"
+              styleName="member-action"
+              disabled={loading}
+            >
+              <h6>Remove</h6>
+            </Button>
+          ) : null}
+          {mode === 'transfer' &&
+          el.row.original.access !== 'owner' &&
+          transferUser === null ? (
+            <Button
+              onClick={() => setTransferUser(el.row.original)}
+              styleName="ownership-button"
+            >
+              Transfer Ownership
+            </Button>
+          ) : null}
+        </>
+      )
+    }
+  ];
+
+  const transferColumns = [
+    memberColumn,
+    {
+      Header: loading ? (
+        <LoadingSpinner
+          placement="inline"
+          className="project-members__loading"
+        />
+      ) : (
+        ''
+      ),
+      accessor: 'username',
+      headerClassName: 'project-members__loading-header',
+      className: 'project-members__cell',
+      Cell: el =>
+        mode === 'transfer' && el.row.original === transferUser ? (
+          <div styleName="confirm-controls">
+            <span>Confirm Ownership Transfer:</span>
+            <Button onClick={confirmTransfer} styleName="ownership-button">
+              Confirm
+            </Button>
+            <Button
+              onClick={() => setTransferUser(null)}
+              color="link"
+              styleName="member-action"
+            >
+              <h6>Cancel</h6>
+            </Button>
+          </div>
+        ) : null
+    }
+  ];
+
+  const isTransferring = mode === 'transfer' && transferUser;
+  const listStyle = `member-list ${
+    isTransferring ? 'transfer-list' : 'addremove-list'
+  }`;
+
+  return (
+    <div styleName="root">
+      <Label className="form-field__label" size="sm">
+        Add Member
+      </Label>
+      <div>
+        <div className="input-group">
+          <div className="input-group-prepend">
+            <Button
+              styleName="add-button member-search"
+              onClick={() => onAdd({ user: selectedUser, access: 'edit' })}
+              disabled={
+                !selectedUser ||
+                loading ||
+                alreadyMember(selectedUser) ||
+                mode === 'transfer'
+              }
+            >
+              Add
+            </Button>
+          </div>
+          <Input
+            list="user-search-list"
+            type="text"
+            onChange={e => userSearch(e)}
+            styleName="member-search"
+            disabled={loading || mode === 'transfer'}
+          />
+          <datalist id="user-search-list">
+            {/* eslint-disable */
+              // Need to replace this component with a generalized solution from FP-743
```

Comment:
This should have a placeholder of `Search by name, email, or username` -- but is our search incapable of handling that? If so, `Search by name` for now?

Reply:
Partial resolution. I've added the ability to search by username as well on the backend. The problem is that using an Input control tied to a datalist only works if the value of the datalist item starts with whatever you're typing in. Example:

I search for "myusername" as the username, and it returns a "User Name (myusername)". Because the value displayed in the datalist is "User Name (myusername)" and what i typed in begins with the letter "m", it doesn't show up as the option to select it.

The solution is to make a dedicated component that does what we want, but it's going to take some effort.